### PR TITLE
Misc July Updates

### LINKED ISSUE #190: Outdated card0 checks
Checking using `card0 == 254`, `card0 == 255`, `card0 == -256` is outdated. rAthena changed the value years ago, and those values must be moved to config and make function/method if want to be checked

### PR DESCRIPTION
**Pull request by cydh**
:heavy_check_mark: Support for Random Option info in all item listings (inventory, storage, cart, guild storage, vending, ~~buyingstore~~)
:heavy_check_mark: Added item list view for vending and buyingstore
:heavy_check_mark: Add more filtering on *rA Logs* Logdata modules
:heavy_check_mark: Moved bound labels to its own file
:heavy_check_mark: Added config for `card0` special values. #194 is added here fixes #190
:heavy_check_mark: Added 'Beloved' label for renamed pet egg (in near future will be added Pet data for name look up)

### CODE REVIEW

**Comment by ecdarreola** [comment]
It's missing js/flux.itemoption.js

**Comment by cydh** [comment]
I will merge this after #210 merged, I need the RespectValidation for validation input in filters

**Comment by Haikenz** [comment]
@cydh This error appears when you see Characters
`PHP Warning:  A non-numeric value encountered in /lib/Flux/Item.php on line 118`

File: modules/vending/index.php [hunk 1]
```diff
@@ -1,27 +1,98 @@
...
+$itemLib = new Flux_Item($server);
+$tempTable = new Flux_TemporaryTable($server->connection, $itemDB, $fromTables);
+
+$item = trim($params->get('item'));
+$refine = trim($params->get('refine'));
+$card = trim($params->get('card'));
+
+$sql_criteria = "";
+$sql_params = array();
+if ($refine) {
```
Comment (cydh) [inline modules/vending/index.php]:
```suggestion
if (preg_match('/^\d+$/', $refine)) {
```

File: modules/vending/index.php [hunk 2]
```diff
@@ -1,27 +1,98 @@
...
+    $sql_criteria .= '`ci`.`refine`=?';
+    $sql_params[] = $refine;
+}
+if ($card) {
```
Comment (cydh) [inline modules/vending/index.php]:
```suggestion
if (preg_match('/^\d+$/', $card)) {
```

File: modules/vending/index.php [hunk 3]
```diff
@@ -1,27 +1,98 @@
...
+    if (count($sql_params))
+        $sql_criteria .= ' AND ';
+    $sql_criteria .= '(`ci`.`card0`=? OR `ci`.`card1`=? OR `ci`.`card2`=? OR `ci`.`card3`=?) ';
+    $sql_params[] = $card;
+    $sql_params[] = $card;
+    $sql_params[] = $card;
+    $sql_params[] = $card;
+}
+if ($item) {
```
Comment (cydh) [inline modules/vending/index.php]:
```suggestion
if (preg_match('/^[\w\'\- ]+$/', $item)) {
```

File: modules/vending/vendors.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+$title = 'Vendors';
+
+$vendor = trim($params->get('vendor'));
+$shopname = trim($params->get('title'));
+$map = trim($params->get('map'));
+
+$sql_criteria = "";
+$sql_params = array();
+if ($vendor) {
```
Comment (cydh) [inline modules/vending/vendors.php]:
```suggestion
if (preg_match('/^[\w\'\- ]+$/', $vendor)) {
```
More allowed char name on server?

File: modules/vending/vendors.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    $sql_criteria .= '`name` LIKE ? ';
+    $sql_params[] = "%$vendor%";
+}
+if ($shopname) {
```
Comment (cydh) [inline modules/vending/vendors.php]:
```suggestion
if (preg_match('/^[\w\'\- ]+$/', $shopname)) {
```

File: modules/vending/vendors.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    if (count($sql_params))
+        $sql_criteria .= ' AND ';
+    $sql_criteria .= '`title` LIKE ? ';
+    $sql_params[] = "%$shopname%";
+}
+if ($map) {
```
Comment (cydh) [inline modules/vending/vendors.php]:
```suggestion
if (preg_match('/^[\w\-]+$/', $map)) {
```

**Comment by douglasparker** [comment]
Is this ready to be merged?

**Comment by Akkarinage** [comment]
We're not merging monster-sized Pull Requests that affects 57 files for 13 different reasons. If anyone wants to, they can separate out the changes if they're still beneficial to FluxCP as individual PRs.

**Pull request closed without merging**